packages/expo-permissions/android/src/main/java/expo/modules/permissions/requesters/SimpleRequester.kt

```diff
@@ -1,32 +1,27 @@
 package expo.modules.permissions.requesters
 
 import android.os.Bundle
-import expo.modules.permissions.EXPIRES_KEY
-import expo.modules.permissions.GRANTED_VALUE
-import expo.modules.permissions.PERMISSION_EXPIRES_NEVER
-import expo.modules.permissions.PermissionsService
-import expo.modules.permissions.STATUS_KEY
-import expo.modules.permissions.UNDETERMINED_VALUE
+import org.unimodules.interfaces.permissions.PermissionsResponse.EXPIRES_KEY
+import org.unimodules.interfaces.permissions.PermissionsResponse.PERMISSION_EXPIRES_NEVER
+import org.unimodules.interfaces.permissions.PermissionsResponse.STATUS_KEY
+import org.unimodules.interfaces.permissions.PermissionsStatus
 
 /**
  * Used for representing CAMERA, CONTACTS, AUDIO_RECORDING, SMS
  */
-class SimpleRequester(private val permissionsService: PermissionsService,
-                      private val permission: String) : PermissionRequester {
-  override fun getAndroidPermissions(): Array<String> = arrayOf(permission)
+class SimpleRequester(vararg val permission: String) : PermissionRequester {
+  override fun getAndroidPermissions(): List<String> = permission.toList()
 
-  override fun getPermission(): Bundle {
+  override fun parseAndroidPermissions(permissionsResponse: Map<String, PermissionsStatus>): Bundle {
     return Bundle().apply {
+      // combined status is equal:
+      // granted when all needed permissions have been granted
+      // denied when all needed permissions have been denied
+      // undetermined if exist permission with undetermined status
       putString(STATUS_KEY, when {
-        permissionsService.isPermissionGranted(permission) -> {
-          GRANTED_VALUE
-        }
-        permissionsService.didAsk(permission) -> {
-          GRANTED_VALUE
-        }
-        else -> {
-          UNDETERMINED_VALUE
-        }
+        getAndroidPermissions().all { permissionsResponse[it] == PermissionsStatus.GRANTED } -> PermissionsStatus.GRANTED.jsString
+        getAndroidPermissions().all { permissionsResponse[it] == PermissionsStatus.DENIED } -> PermissionsStatus.DENIED.jsString
```

Comment:
previously you create local variables for these case matchers 😜 może zachowaj spójność? 🤔